src/NumericalAlgorithms/Spectral/Chebyshev.cpp

```diff
@@ -0,0 +1,121 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+
+#include <cmath>
+#include <cstddef>
+// IWYU pragma: no_include <type_traits>
+#include <utility>
+
+#include "DataStructures/DataVector.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+
+namespace Spectral {
+
+// Algorithms to compute Chebyshev basis functions
+// These functions specialize the templates declared in `Spectral.cpp`.
+
+/// \cond
+template <Basis>
+DataVector compute_basis_function_values(size_t, const DataVector&) noexcept;
+template <>
+DataVector compute_basis_function_values<Basis::Chebyshev>(
+    const size_t k, const DataVector& x) noexcept {
+  // Algorithm 21 from Kopriva's book, p. 60
+  // Since the trigonometric form is expensive to compute we make use of it only
+  // for large k. It remains to be determined what is a reasonable crossover
+  // point between the two forms, e.g. depending on compiler settings.
+  static constexpr size_t chebyshev_recursion_to_trigonometry_crossover = 10;
```

Comment:
Kopriva comments on this in the book (not in these exact words). Up to some `k` the recursion is faster than computing the `cos` and `acos`. At which `k` it becomes more efficient to compute the trigonometric form, as opposed to performing the recursion, varies a lot, so Kopriva recommends testing this for every compiler and computer architecture. Since this is not of high priority right now I added the comment that this is to be determined.
Added a reference to the section in the book where this is discussed now, too.

Reply:
The result of this function is cached.  Marginal speed improvements are not worth complicating the logic here.  Just pick one method.

Reply:
I agree. For the (small) values of k we will be using, recursion is the way to go.

Reply:
sounds good 👍